controllers/account.go

```diff
@@ -90,6 +92,13 @@ func (c *ApiController) Signup() {
 		return
 	}
 
+	for key := range authForm.CustomFields {
+		if strings.Contains(strconv.Itoa(key), " ") || regexp.MustCompile(`[!@#$%^&*(),.?":{}|<>]`).MatchString(strconv.Itoa(key)) {
```

Comment:
compile regex when initial like 
https://github.com/casdoor/casdoor/blob/a330fbc11fd24e6150e6d911f7d230d4ce5d47bc/util/validation.go#L33-L38

Reply:
and have you test this function? authForm.CustomFields looks like a string and you just checking index `01234567...`

Reply:
i don't think this way can validate user's input

Reply:
> i don't think this way can validate user's input
I modified the code. The previous code did have some problems. I also adjusted it. Please review it.

Reply:
> > i don't think this way can validate user's input

I modified the code. The previous code did have some problems. I also adjusted it. Please review it.